edward/criticisms/evaluate.py

```diff
@@ -330,9 +380,10 @@ def mean_squared_error(y_true, y_pred):
     y_pred: tf.Tensor.
       Tensors of same shape and type.
   """
-  return tf.reduce_mean(tf.square(y_pred - y_true))
+  return tf.reduce_mean(tf.square(y_pred - y_true), axis=-1)
```

Comment:
We always want metrics to return a scalar. This doesn't happen if, say, we're doing matrix regression where the output is a matrix, and we predict using a matrix and calculate MSE with this function.

Reply:
In fact, given the decorator, it always will return a scalar (unless `average=None` is specified). This logic plays out on lines 23 and 30 of "edward/util/metrics.py".

Reply:
Perfect!